src/content/docs/en/recipes/build-forms.mdx

```diff
@@ -0,0 +1,200 @@
+---
+title: Build HTML Forms in Astro Pages
+description: Learn how to build HTML forms and handle submissions in your frontmatter
+i18nReady: false
+type: recipe
+---
+
+In SSR mode, Astro pages can both display and handle forms. In this recipe, you'll use a standard HTML form to submit data to the server. Your frontmatter script will handle the data on the server, sending no JavaScript to the client.
+
+## Recipe
+
+1. Create or identify a `.astro` page which will contain your form and your handling code. For example, you could add a registration page:
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    ```
+
+2. Add a `<form>` tag with some inputs to the page. Each input should have a `name` attribute that describes the value of that input. 
+
+    Be sure to include a `<button>` or `<input type="submit">` element to submit the form.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
+        <input type="text" name="username" />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" />
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+
+3. Use [validation attributes](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation) to provide basic client-side validation that works even if JavaScript is disabled. 
+
+    In this example,
+    - `required` prevents form submission until the field is filled. 
+    - `minlength` sets a minimum required length for the input text.
+    - `type="email"` also introduces validation that will only accept a valid email format.
+
+    ```astro title="src/pages/register.astro"
+    ---
+    ---
+    <h1>Register</h1>
+    <form>
+      <label>
+        Username:
+        <input type="text" name="username" required />
+      </label>
+      <label>
+        Email:
+        <input type="email" name="email" required />
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" required minlength="6" />
+      </label>
+      <button>Submit</button>
+    </form>
+    ```
+    
+    :::tip
+    You can add custom validation logic, and validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
```

Comment:
The use of "validation logic" almost back-to-back sounded a bit confusing for me, what about reworking this a bit? - Not sure that "custom validation logic that refers to multiple fields" is misleading, as in, you can't have both, so feel free to ignore and propose something else here.
```suggestion
    You can add custom validation logic that refers to multiple fields, using a `<script>` tag and the [Constraint Validation API](https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#complex_constraints_using_the_constraint_validation_api).
```

Reply:
I like Yan's proposal! @Jutanium happy for you to approve this one, but... extra comma here because of deleting the clause, I think?

Reply:
I used the ol' "edit comment and commit" trick 😬